### PR TITLE
chore(routing): upgrade to ui-router 1.0.rc1

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Still need to do some testing, but I'm sure it's fine, it's just the router and we barely do anything unusual with it.

I did not go as far as I could have with this PR in refactoring: the filter models could probably extend from a base class instead of the `asFilterModel` assignment, the parameter management could almost certainly be cleaned up more. I wanted to make as few changes as possible and avoid anything too drastic when something inevitably breaks.

@christopherthielen PTAL

### CODE REVIEW

File: app/scripts/modules/core/cluster/filter/clusterFilter.model.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      if (oldBase === newBase) {
+        this.mostRecentParams = newQuery ? this.urlParser.parseQueryString(newQuery) : {};
+      } else {
+        this.mostRecentParams = oldQuery ? this.urlParser.parseQueryString(oldQuery) : {};
+      }
+    });
+
+    this.$rootScope.$on('$stateChangeStart', this.handleStateChangeStart);
+
+    this.$rootScope.$on('$stateChangeSuccess', (_event: IAngularEvent, toState: State, toParams: StateParams, fromState: State) => {
```
Comment (jrsquared) [inline app/scripts/modules/core/cluster/filter/clusterFilter.model.ts]: Do these `$rootScope.$on('$stateChange*...` listeners still make sense in `ui-router@1.0` world?

File: app/scripts/modules/core/cluster/filter/clusterFilter.service.spec.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import {CLUSTER_FILTER_SERVICE, ClusterFilterService} from 'core/cluster/filter/
 import {CLUSTER_SERVICE} from 'core/cluster/cluster.service';
 import {Application} from 'core/application/application.model';
 import {APPLICATION_MODEL_BUILDER, ApplicationModelBuilder} from 'core/application/applicationModel.builder';
+import { CLUSTER_FILTER_MODEL} from './clusterFilter.model';
```
Comment (jrsquared) [inline app/scripts/modules/core/cluster/filter/clusterFilter.service.spec.ts]: mixed spaces.

File: app/scripts/modules/core/cluster/filter/multiselect.model.js [hunk 3]
```diff
@@ -3,11 +3,12 @@
 import {Subject} from 'rxjs';
 
 let angular = require('angular');
+import { CLUSTER_FILTER_MODEL} from './clusterFilter.model';
```
Comment (jrsquared) [inline app/scripts/modules/core/cluster/filter/multiselect.model.js]: mixed spaces

File: app/scripts/modules/core/cluster/onDemand/onDemandClusterPicker.component.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import {IComponentController, IComponentOptions, IScope, module} from 'angular';
 import {Application} from 'core/application/application.model';
 import {IClusterSummary} from 'core/domain/ICluster';
+import { CLUSTER_FILTER_MODEL} from '../filter/clusterFilter.model';
```
Comment (jrsquared) [inline app/scripts/modules/core/cluster/onDemand/onDemandClusterPicker.component.ts]: mixed spaces

File: app/scripts/modules/core/loadBalancer/loadBalancer.directive.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 import _ from 'lodash';
```
Comment (jrsquared) [inline app/scripts/modules/core/loadBalancer/loadBalancer.directive.js]: fix all the other `_.` imports so we can remove this.

File: app/scripts/modules/core/task/taskExecutor.ts [hunk 6]
```diff
@@ -6,7 +6,7 @@ export interface IJob {
   account?: string;
   application?: any;
   providerType?: string;
-  source?: string;
+  source?: any;
```
Comment (jrsquared) [inline app/scripts/modules/core/task/taskExecutor.ts]: 😢

File: app/scripts/modules/netflix/fastProperties/fastProperties.states.ts [hunk 7]
```diff
@@ -1,17 +1,12 @@
 import {module} from 'angular';
+import { StateParams } from 'angular-ui-router';
+
```
Comment (jrsquared) [inline app/scripts/modules/netflix/fastProperties/fastProperties.states.ts]: so much whitespace... what is this, material design?

**Comment by anotherchrisberry** [comment]
Closed in favor of https://github.com/spinnaker/deck/pull/3623

**Pull request closed without merging**